test/generate-testcase.js

```diff
@@ -165,3 +166,23 @@ function runReadability(source, destPath, metadataDestPath) {
   });
 }
 
+if (process.argv.length < 3) {
```

Comment:
The problem with the metadata files is, I suspect, a few lines above here (github's not letting me comments on lines not changed in the PR, sigh...) - after writing the metadata file, the code calls `process.exit(0)`, which in the `all` case will now happen N times for the N things we process - I bet we cut writing the output short. If calling `exit` explicitly is necessary, we'll have to track completion of all the items and then call it; otherwise we may be able to just remove it?

Reply:
Yup, that was it; everything seems to work after removing it.